pattern/constraint/thing/ValueConstraint.java

```diff
@@ -157,7 +157,7 @@ public java.lang.String toString() {
         return owner.toString() + SPACE + predicate.toString() + SPACE + value.toString();
     }
 
-    public boolean inconsistentWith(ValueConstraint<?> valueConstraint) {
+    public <TYPE> boolean inconsistentWith(ValueConstraint<TYPE> valueConstraint) {
```

Comment:
type parameter lets us access a `ValueConstraint`'s value encoding and actual value bound to the same type